### PR TITLE
postgres volume driver

### PR DESCRIPTION
**Pull request by joshuacox**
Added volume driver selection for postgres, and the possibility to choose hosts by label, and a little bit of cleanup.

It should be noted that I am seeing issues with NFS and databases, and I'd like as much feedback as possible on this:
https://forums.rancher.com/t/rancher-nfs-and-databases/7094

Is it even a good idea to use `rancher-nfs` or anything other than `local` for a database?

### CODE REVIEW

File: templates/postgres/1/docker-compose.yml.tpl [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    labels:
+      io.rancher.container.start_once: true
+    {{- if ne .Values.host_label ""}}
+      io.rancher.scheduler.affinity:host_label: ${host_label}
+    {{- end}}
+    volumes:
+      - pgdata:/var/lib/postgresql/data/pgdata
+
+  postgres:
+    image: postgres:latest
```
Comment (rawmind0) [inline templates/postgres/1/docker-compose.yml.tpl]: Please don't use latest, use a concrete version for postgres image..
```
-    image: postgres:latest
+    image: postgres:<version>
```

**Comment by joshuacox** [comment]
How if I make it an `enum` choice?

**Comment by rawmind0** [comment]
@joshuacox , nice to use a enum to select postgres version. :)

One more thing, could you please edit config.yml file to set new ```version: v0.0.2``` as default one??

**Comment by joshuacox** [comment]
you bet, and as an aside note, I have not had a single issue with this and rancher-nfs, I think when I had problems with postgres previously it was a fluke, or convoy-nfs versus rancher-nfs.  And unrelated to the issues I am having with mongo.

**Comment by rawmind0** [comment]
I've answered to your post https://forums.rancher.com/t/rancher-nfs-and-databases/7094 :)

Thanks for your commit...

LGTM...

**Pull request merged**